lib/diaspora_federation/salmon/xml_payload.rb

```diff
@@ -85,19 +85,32 @@ def self.entity_class_name(term)
       # @param [Nokogiri::XML::Element] node xml nodes
       # @return [Entity] instance
       def self.populate_entity(klass, node)
-        data = {}
-        klass.class_props.each do |prop_def|
-          name = prop_def[:name]
-          type = prop_def[:type]
-
-          if type == String
-            data[name] = parse_string_from_node(prop_def[:xml_name], node)
-          elsif type.instance_of?(Array)
-            data[name] = parse_array_from_node(type, node)
-          elsif type.ancestors.include?(Entity)
-            data[name] = parse_entity_from_node(type, node)
+        # Build a hash of attributes basing on XML tree. If elements are known in "props" they respect the Entity logic.
+        # All other elemnts are respected and attached to resulted hash as string.
+        # It is intended to build a hash invariable of an Entity definition, in order to support receiving objects
+        # from the future versions of Diaspora, where new elements may have been added.
+        xml_names = klass.class_props.map {|prop_def| prop_def[:xml_name] }
+
+        data = node.element_children.map { |child|
+          xml_name = child.name.to_sym
```

Comment:
You don't want the `to_sym` for all properties, only for the known ones. And you don't need the name from unknown properties anyway, because you only need the value to verify the signature.